test-suite-data.json

```diff
@@ -706,5 +706,20 @@
     "qualifiers": null,
     "subpath": null,
     "is_invalid": true
+  },
+  {
+    "purl": "pkg:yocto/openembedded-core/glibc@2.35&repository_url=https%3A%2F%2Fgit.openembedded.org%2Fopenembedded-core&layer_version=kirkstone",
```

Comment:
`core` instead of `openembedded-core`

Reply:
I'm not quite sure here...
`BBLAYERS_LAYERINDEX_NAME_core = "openembedded-core"`

Reply:
Also this says that the name should be `openembedded-core` and not `core`: https://github.com/yoctoproject/poky/blob/master/meta/classes/toaster.bbclass#L45

Reply:
He is right, because we use `BBFILE_COLLECTIONS`:  https://git.openembedded.org/openembedded-core/tree/meta/conf/layer.conf#n6

Reply:
> I'm not quite sure here... `BBLAYERS_LAYERINDEX_NAME_core = "openembedded-core"`

That's the name known to the layer index, which isn't necessarily the "layer name". Not all layers are published to the layer index, so using that name is probably unwise (unless it's completely optional to specify it....). Bitbake knows this layer by the `core` name (as is apparent in the suffix of the `BBLAYERS_LAYERINDEX_NAME` variable _et. al._), and that named is required (as much as anything) and needs to be unique among all layers. Finally, bitbake presents this name to the recipe code when it is parsed as `FILE_LAYERNAME`, so we can construct a valid PURL with a minimal: `pkg:yocto/${FILE_LAYERNAME}/${BPN}@${PV}`